OpenSim/Common/ComponentList.h

```diff
@@ -76,6 +78,21 @@ class ComponentFilterMatchAll : public ComponentFilter {
         return new ComponentFilterMatchAll(*this);
     }
 };
+
+/** A component is considered a match if its full path name matches the 
+provided regular expression. */
+class ComponentFilterFullPathNameRegex : public ComponentFilter {
+public:
+    ComponentFilterFullPathNameRegex(const std::regex& expr) : _expr(expr) {}
```

Comment:
Good point about regex. I agree: simpler is best; we can add regex in the future if desired/necessary. (We may then also want to prohibit special characters in Component names as well.)